ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/NonStakeholderInformeesITSpec.scala

```diff
@@ -0,0 +1,174 @@
+
+class NonStakeholderInformeesITSpec
+    extends AsyncFlatSpec
+    with SandboxFixture
+    with SuiteResourceManagementAroundAll
+    with Matchers
+    with AppendedClues
+    with OptionValues {
+
+  it should "divulge events" in {
+    val expectedTemplateNames = Set("Foo1", "Divulger")
+    val submissionConfig = WorkflowConfig.FooSubmissionConfig(
+      numberOfInstances = 100,
+      numberOfObservers = 1,
+      numberOfDivulgees = 3,
+      uniqueParties = false,
+      instanceDistribution = List(
+        WorkflowConfig.FooSubmissionConfig.ContractDescription(
+          template = "Foo1",
+          weight = 1,
+          payloadSizeBytes = 0,
+        )
+      ),
+      nonConsumingExercises = None,
+      consumingExercises = None,
+    )
+    for {
+      ledgerApiServicesF <- LedgerApiServices.forChannel(
+        channel = channel,
+        authorizationHelper = None,
+      )
+      apiServices: LedgerApiServices = ledgerApiServicesF("someUser")
+      submitter = CommandSubmitter(
+        names = new Names(),
+        benchtoolUserServices = apiServices,
+        adminServices = apiServices,
+        metricRegistry = new MetricRegistry,
+        metricsManager = NoOpMetricsManager(),
+      )
+      (signatory, observers, divulgees) <- submitter.prepare(submissionConfig)
+      tested = new FooSubmission(
+        submitter = submitter,
+        maxInFlightCommands = 1,
+        submissionBatchSize = 5,
+        submissionConfig = submissionConfig,
+        signatory = signatory,
+        allObservers = observers,
+        allDivulgees = divulgees,
+      )
+      _ <- tested.performSubmission()
+      (treeResults_divulgee0, flatResults_divulgee0) <- observeAllTemplatesForParty(
+        party = divulgees(0),
+        apiServices = apiServices,
+        expectedTemplateNames = expectedTemplateNames,
+      )
+      (treeResults_divulgee1, flatResults_divulgee1) <- observeAllTemplatesForParty(
+        party = divulgees(1),
+        apiServices = apiServices,
+        expectedTemplateNames = expectedTemplateNames,
+      )
+      (treeResults_observer0, flatResults_observer0) <- observeAllTemplatesForParty(
+        party = observers(0),
+        apiServices = apiServices,
+        expectedTemplateNames = expectedTemplateNames,
+      )
+      (treeResults_signatory, _) <- observeAllTemplatesForParty(
+        party = signatory,
+        apiServices = apiServices,
+        expectedTemplateNames = expectedTemplateNames,
+      )
+    } yield {
+      // Creates of Foo1 are divulged to "divulgee" party,
+      // thus, they are visible on transaction trees stream but absent from flat transactions stream.
+      {
+        // Divulge0
+        val treeFoo1 = treeResults_divulgee0.numberOfCreatesPerTemplateName("Foo1")
+        val flatFoo1 = flatResults_divulgee0.numberOfCreatesPerTemplateName("Foo1")
+        treeFoo1 shouldBe 100 withClue ("number of Foo1 contracts visible to divulgee0 on tree transactions stream")
```

Comment:
My understanding of this is that the tree events are observed because the divulgee witnesses an (non-consuming) execution of a contract on which they are a signatory.

The flat events are not observed because:
- The flat event stream does not contain (non-consuming) executions 
- The CreateEvent is not visible to the divulgee.  

I'm not sure I see this as a divulgence issue as much as a difference between the transaction stream types.

Reply:
If the Divulgence contracts contain a mixed set of divulgees then how is it that divulgee0 observed 100/100 of the executions ?

Reply:
> My understanding of this is that the tree events are observed because the divulgee witnesses an (non-consuming) execution of a contract on which they are a signatory.

That reasoning would apply to exercising a nonconsuming `DivulgeImmedate` on `Divulger` contracts. 
However the divulged events are creations of `Foo1`.

For the `Foo1` party a divulgee witnesses creation of `Foo1` solely because the creation event is a sub-action of an action (a choice) on which divulgee is a signatory.

The flat transactions stream doesn't contain creations of `Foo1` because:
- divulgee is not a stakeholder on `Foo1`
- flat transaction's use `flat_event_witnesses` and for creations it's  set to `flat_event_witnesses = stakeholders,` in the code:  https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/store/backend/UpdateToDbDto.scala#L160